supergloo/docs/mesh/mesh-discovery.md

```diff
@@ -0,0 +1,67 @@
+---
+title: "Mesh Discovery"
+weight: 2
+---
+
+# Overview
+
+Mesh discovery is the ability to discovery service meshes which are running in the cluster to which mesh
+discovery is deployed. This capability is currently shipped with supergloo by default, but in the future
+will be available as a standalone feature.
+
+Currently supported meshes for Discovery:
+
+- Istio
+
+# Architecture
+
+Mesh discovery, similar to other solo.io projects, uses an event loop based architecure to watch kubernetes 
+resources to create/update those resources. In this case we are interested in three resources:
+
+* [Mesh](../v1/github.com/solo-io/supergloo/api/v1/mesh.proto.sk.md)
+* [Install](../v1/github.com/solo-io/supergloo/api/v1/install.proto.sk.md)
+* [Pod](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#pod-v1-core) (kubernetes resource)
+
+More information about these resources can be found by clicking the links above. 
+
+At it's core mesh discovery monitors these resources and takes actions based on certain heuristics. These heuristics 
+vary slightly by the type of mesh we are trying to discover, but the concept is similar. For example, in the case of istio: 
+mesh discovery watches for the existence of istio-pilot pods and creates mesh crds based on the internals of those pods.
```

Comment:
for the existence of a deployment named `istio-pilot` and discovers the deployed Istio version based on Pilot's image tag